File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/strategy/MainStrategy.java
package designPatterns.behavioral.strategy;

public class MainStrategy {
  public static void main(String[] args) {
    Context context = new Context(new Operation.Add());
    System.out.println("10 + 5 = " + context.executeStrategy(10, 5));

    context = new Context(new Operation.Subtract());
    System.out.println("10 - 5 = " + context.executeStrategy(10, 5));

    context = new Context(new Operation.Multiply());
    System.out.println("10 * 5 = " + context.executeStrategy(10, 5));
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/strategy/MainStrategy.kt
package behavioral.strategy

fun main() {
  var context = Context(Operation.Add())
  println("10 + 5 = ${context.executeStrategy(10, 5)}")

  context = Context(Operation.Subtract())
  println("10 - 5 = ${context.executeStrategy(10, 5)}")

  context = Context(Operation.Multiply())
  println("10 * 5 = ${context.executeStrategy(10, 5)}")
}